pkg/devicerepository/store/bleve/init.go

```diff
@@ -203,11 +213,67 @@ func (c Config) Initialize(ctx context.Context, lorawanDevicesPath string, overw
 			}
 		}
 
+		// Add the end device provisioning info to the index.
+		for _, model := range models.Models {
+			profiles, err := s.GetEndDeviceProfiles(store.GetEndDeviceProfilesRequest{
+				ModelID: model.ModelId,
+			})
+			if err != nil && !errors.IsNotFound(err) {
+				return err
+			}
+			if profiles == nil {
+				continue
+			}
+
+			for _, profile := range profiles.Profiles {
+				vendorID := strconv.Itoa(int(brand.LoraAllianceVendorId))
+				vendorProfileID := strconv.Itoa(int(profile.VendorProfileID))
+
+				template, err := s.GetTemplate(&ttnpb.GetTemplateRequest{
+					EndDeviceProfileIds: &ttnpb.GetTemplateRequest_EndDeviceProfileIdentifiers{
+						VendorId:        brand.LoraAllianceVendorId,
+						VendorProfileId: profile.VendorProfileID,
```

Comment:
The `VendorProfileID` field is removed from the Profile. This is now instead in the `index.yaml` of each vendor. I think we need to account for this also in the other parts of the code where we assume that `vendorProfileID` is in the profile.